mod/01_filter/set/abstract/filter.rb

```diff
@@ -0,0 +1,232 @@
+  "metric"
+end
+
+def params_keys
+  %w(metric designer wikirate_topic project year)
+end
+
+def get_query params={}
+  filter = params_to_hash params_keys
+  search_args = search_wql filter
+  sort_by search_args, Env.params["sort"] if sort?
+  params[:query] = search_args
+  super(params)
+end
+
+def item_cards params={}
+  s = query(params)
+  raise("OH NO.. no limit") unless s[:limit]
+  query = Query.new(s, comment)
+  shift_sort_table query
+  query.run
+end
+
+def shift_sort_table query
+  if sort? && shift_sort_table?
+    # sort table alias always stick to the first table,
+    # but I need the next table
+    sort = query.mods[:sort].scan(/c([\d+]).db_content/).last.first.to_i + 1
+    query.mods[:sort] = "c#{sort}.db_content"
+  end
+end
+
+def sort_by wql, sort_by
+  wql[:sort_as] = "integer"
+  wql[:dir] = "desc"
+  wql[:sort] = {
+    right: (sort_by || default_sort_by_key), right_plus: "*cached count" }
+end
+
+def virtual?
+  true
+end
+
+def raw_content
+  %(
+    {
+      "name":"dummy"
+    }
+  )
+end
+
+def params_to_hash params
+  params.each_with_object({}) do |param, hash|
+    if (val = Env.params[param])
+      hash[param.to_sym] = val
+    end
+  end
+end
+
+def search_wql opts, return_param=nil
+  wql = { type_id: WikirateCompanyID }
+  wql[:return] = return_param if return_param
+  filter_by_name wql, opts[:company] if opts[:company].present?
+  filter_by_industry wql, opts[:industry] if opts[:industry].present?
+  filter_by_project wql, opts[:project] if opts[:project].present?
+  wql
+end
+
+def filter_by_name wql, name
+  return unless name.present?
+  wql[:name] = ["match", name]
+end
+
+def filter_by_project wql, project
+  wql[:referred_to_by] = { left: { name: project } }
+end
+
+def filter_by_industry wql, industry
+  wql[:left_plus] = [
+    format.industry_metric_name,
+    { right_plus: [
+      format.industry_value_year,
+      { right_plus: ["value", { eq: industry }] }
+    ] }
+  ]
+end
+
+format :html do
+  def industry_metric_name
+    "Global Reporting Initiative+Sector Industry"
+  end
+
+  def industry_value_year
+    "2015"
+  end
+
+  def page_link_params
+    []
+  end
+
+  view :no_search_results do |_args|
+    %(
+      <div class="search-no-results">
+        No result
+      </div>
+    )
+  end
+
+  view :filter_form do |args|
+    formgroups = args[:formgroups] || [:name_formgroup]
+    html = formgroups.map do |fg|
+      optional_render(fg, args)
+    end
+    content = output(html)
+    action = card.left.name
+    %( <form action="/#{action}" method="GET">#{content}</form>)
+  end
+
+  def page_link text, page, _current=false, options={}
+    @paging_path_args[:offset] = page * @paging_limit
+    filter_args = {}
+    page_link_params.each do |key|
+      filter_args[key] = params[key] if params[key].present?
+    end
+    options[:class] = "card-paging-link slotter"
+    options[:remote] = true
+    link_to raw(text), path(@paging_path_args.merge(filter_args)), options
+  end
```

Comment:
as noted in the other PR, seems very odd to have this here.  definitely need a comment.